### DISCUSSION TITLE
Issues renaming files

### DISCUSSION THREAD

**Discussion by tamorgen** [gboudreau/Greyhole#332] [General]
I've had an issue with renaming files after the file in the landing zone is copied to the drive, where I simply can't do it.  An example would be I copy a TV series to a landing zone, and I have to go back through and change episode numbers.  What will happen is if the file has already been copied over, it'll rename the file back to the original file.  This is driving me a bit nuts.  

l'm running 0.15.22-1 on Fedora 40 Server

**Comment by gboudreau** [comment]
Hi,

Are you saying you are renaming files directly in the LZ? You should never do that. All file changes should occur through Samba, never directly in the LZ.

If you are indeed always working through Samba:
- When does Greyhole rename the file back to the original filename?
- Can you reproduce this issue, and find the Greyhole daemon logs that show the file being renamed back to the original filename?
- Also find the logs that show the file being renamed initially.

Cheers.

**Comment by tamorgen** [comment]
No, I'm doing it through Samba.  When I do it through Samba, it changes the file name in the LZ, but not on the actual drive.  I try and run fsck, and that's when it changes the LZ file name back.

I'm having problems finding the instance in the /var/log/greyhole.log file, however I did see this:

Sep 03 01:16:49 WARN daemon:   Greyhole VFS module (/usr/lib64/samba/vfs/greyhole.so) seems to be missing some required libraries. If you have issues connecting to your Greyhole-enabled shares, try to compile a new VFS module for Samba by running this command: /usr/share/greyhole/build_vfs.sh current

I've seen this before, and tried recompiling.  Could this cause the behavior I'm seeing?

**Comment by tamorgen** [comment]
I also found this in the logs from this morning, after the server did it's daily updates/reboot:

Sep 04 01:25:11 WARN read_smb_spool: PHP Warning [2]: Error while sending QUERY packet. PID=157883 in /usr/bin/greyhole on line 689; BT: greyhole[L8015] run() => greyhole[L7333] parse_samba_spool() => greyhole[L4277] acquireLock(read_smb_spool,5) => greyhole[L764] getFirstValue(SELECT GET_LOCK(:name, :timeout),array (  'name' => 'read_smb_spool',  'timeout' => 5,)) => greyhole[L721] getFirst(SELECT GET_LOCK(:name, :timeout),array (  'name' => 'read_smb_spool',  'timeout' => 5,)) => greyhole[L713] execute(SELECT GET_LOCK(:name, :timeout),array (  'name' => 'read_smb_spool',  'timeout' => 5,)) => greyhole[L689] execute()
Sep 04 01:25:11 ERROR read_smb_spool: PHP Fatal Error: Uncaught Exception: SQLSTATE[HY000]: General error: 2006 MySQL server has gone away in /usr/bin/greyhole:703

I just ran the build_vfs.sh current command and it completed successfully.

I restarted both samba and greyhole, and I didn't see this warning pop up, so maybe the build command fixed the issue.  I'll have to test renaming a file this evening when I can access the SMB shares.

**Comment by gboudreau** [comment]
If the files are not renamed in the storage pool drives, that means the Greyhole daemon didn't "see" the files being renamed.
When you rename a file, make sure it uses the locally-mounted share path, and look in the Greyhole daemon log.
You should see (if you have leo_level set to DEBUG or more verbose, in `/etc/greyhole.conf`) a file operation for this, and the daemon should rename all the files in the storage pool.

**Reply by tamorgen** [reply]
Thanks, I'll try again this afternoon when I get home.   I can't access the SMB shares remotely.  Hopefully the rebuild fixed the issue.

**Comment by tamorgen** [comment]
Alright, I kept a terminal window up while renaming the file, and did a tail -f on /var/logs/greyhole.log, and the only thing it's showing is:

Sep 04 17:36:42 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:36:52 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:37:02 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:37:12 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:37:22 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:37:32 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:37:42 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:37:52 DEBUG sleep: Nothing to do... Sleeping.
Sep 04 17:38:02 DEBUG sleep: Nothing to do... Sleeping.

In a seperate window, I can see that I renamed the lz file, but it did not rename the file in the storage pool.

t*****@*******:/var/hda/files/lz/tv series/B********/Season 1$ ls -al
total 4
drwxrwxr-x. 2 t****** users 4096 Sep  4 17:36  .
drwxrwxr-x. 3 t****** users   39 Sep  3 20:23  ..
lrwxrwxrwx  1 root          root   101 Sep  4 08:08 'B******* s1e0.mkv' -> '/var/hda/files/drives/drive3/gh/TV Series/B********/Season 1/B********* s1e1.mkv'

**Comment by gboudreau** [comment]
Please show the terminal & command you used to rename the file.
If you worked directly in `/var/hda/files/lz`, that is your problem. You need to rename the file in `/mnt/samba/` (or equivalent).

**Reply by tamorgen** [reply]
I didn't use terminal to rename the file.  I used a Finder window on my Mac.

I just used an ssh terminal session to ls -al to see the what was happening in the  LZ, and another to tail the logs.

**Comment by gboudreau** [comment]
What happens in the greyhole daemon log, when you add a file into that share? Is it handled as soon as its added (within 10 secs)?

Do you have your Samba shares mounted locally on your server, in `/mnt/samba`? If so, please try to use that path in a terminal to rename a file, and see if that works or not.

Please run this, and send me the URL you receive: https://github.com/gboudreau/Greyhole/wiki/GetHelp#gather-system-information  
I'll see if maybe the share in question (`TV Series`) is misconfigured.

While renaming, try to tail all the Samba logs: `sudo tail -f /var/log/samba/*`  
Are there any errors in there, when renaming?

**Comment by tamorgen** [comment]
I've attached the logs while adding a file to the share.  

[smb logs.txt](https://github.com/user-attachments/files/16898894/smb.logs.txt)


t********@galactica:/var/hda/files/lz/tv series/B*********/Season 1$ ls -al
total 20
drwxrwxr-x. 2 tmorgenthaler users         4096 Sep  5 14:32  .
drwxrwxr-x. 3 tmorgenthaler users           39 Sep  3 20:23  ..
-rw-r--r--  1 ********* ********* 8607 Mar 19  2011  vw-nut.jpeg

I don't see a link being set up from the LZ to the share drives.

Running the command you provided doesn't seem to be working.....

t********@galactica:~$ curl -Ls https://bit.ly/gh-infos | sudo sh
ls: cannot access '/mnt/samba/': No such file or directory
/etc/init.d/greyhole: line 121: killproc: command not found
/etc/init.d/greyhole: line 126: failure: command not found
/etc/init.d/greyhole: line 95: daemon: command not found
/etc/init.d/greyhole: line 100: failure: command not found
Here's the URL you will need to give the to person who's helping you:
<!doctype html>
<html lang=en>
<head>
    <meta charset=utf-8>
    <title>&nbsp;</title>
</head>
<style>
    body {
        font-family: monospace;
	margin: 2em;
    }
</style>
<body>
    <p>ix.io is taking a break &#127867;</p>
    <img src="/underconstruction.gif" width="200px">
</body>
</html>

This used to all work.  I'm not sure what is going on.

**Comment by gboudreau** [comment]
If files are not processed, when they are added onto your share, your Greyhole VFS is not working.
My guess is that your `smb.conf` is not correctly activating the Greyhole VFS module for your share.

See https://github.com/gboudreau/Greyhole/blob/master/USAGE#L55

You can use the `testparm` command to see your parsed samba config. Confirm the you have a `	vfs objects = greyhole` line for this specific share.

**Reply by tamorgen** [reply]
You're right, it seems as if the following two lines were missing in the smb.conf file.  I'm not sure if they got removed by an update, or something else.  I added the lines to each share and now renaming appears to work properly.

        dfree command = /usr/bin/greyhole-dfree
        vfs objects = greyhole

Thanks!
